lib/lightning/accounts.ex

```diff
@@ -492,44 +494,55 @@ defmodule Lightning.Accounts do
 
   ## Examples
 
-      iex> deliver_update_email_instructions(user, current_email, &Routes.user_update_email_url(conn, :edit, &1))
+      iex> deliver_update_email_instructions(user, new_email, &Routes.user_update_email_url(conn, :edit, &1))
       {:ok, %{to: ..., body: ...}}
 
   """
   def deliver_update_email_instructions(
         %User{} = user,
-        current_email,
+        new_email,
```

Comment:
Instead of passing in just the email, could we pass in an unpersisted User struct. That way the user will get the email with their name on it as well instead of just `their@email.com`.

Reply:
@stuartc I don't quite understand relevancy of this comment (sorry I might be missing something). This function is a wrapper around the `UserNotification.deliver_update_email_instructions/2` which takes a User struct and a URL. The `Accounts.deliver_update_email_instructions/3` function takes a User struct, an email string, and a URL string, build the token, and assigns the new email to the User struct before calling the `UserNotification.deliver_update_email_instructions/2`. So I don't understand why we need to have another User struct.

```
  def deliver_update_email_instructions(
        %User{} = user,
        new_email,
        update_email_url_fun
      )
      when is_function(update_email_url_fun, 1) do
    {encoded_token, user_token} =
      UserToken.build_email_token(user, "change:#{user.email}", new_email)

    Repo.insert!(user_token)

    UserNotifier.deliver_update_email_warning(
      user,
      new_email
    )

    UserNotifier.deliver_update_email_instructions(
      %User{
        email: new_email,
        first_name: user.first_name,
        last_name: user.last_name
      },
      update_email_url_fun.(encoded_token)
    )
  end
```